rust/mlt/src/decoder/tracked_bytes.rs

```diff
@@ -0,0 +1,66 @@
+use std::ops::{Deref, DerefMut};
+
+use bytes::{Buf, Bytes};
+
+pub struct TrackedBytes(usize, Bytes);
+
+impl TrackedBytes {
+    pub fn new(data: Vec<u8>) -> Self {
+        let bytes = Bytes::from(data);
```

Comment:
Bytes already has a bunch of `from` implementations, no need to hard-code one of them. This code would allow calling `new` with anything that can be converted into bytes.

```suggestion
    pub fn new<B: Into<Bytes>>(data: B) -> Self {
        let bytes = data.into();
```

Reply:
Thank you so much for pointing this out. It is indeed more generic now.

Reply:
hehe, well... that's the whole idea of `generics` :D